libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -86,8 +93,13 @@ void AC_Circle::init()
     const Vector3f& stopping_point = _pos_control.get_pos_target();
 
     // set circle center to circle_radius ahead of stopping point
-    _center.x = stopping_point.x + _radius * _ahrs.cos_yaw();
-    _center.y = stopping_point.y + _radius * _ahrs.sin_yaw();
+    if (_radius_used) {
+        _center.x = stopping_point.x + _radius * _ahrs.cos_yaw();
+        _center.y = stopping_point.y + _radius * _ahrs.sin_yaw();
+    } else {
+        _center.x = stopping_point.x;
+        _center.y = stopping_point.y;
+    }
```

Comment:
@amilcarlucas san. Comments thanks.
The original circle mode is
1. Draw an arc from the base point with the radius specified in the head direction of the fuselage.
2. If the radius is 0, rotate the aircraft at the base point. I call it a panorama here.
I think it is better to have 1 and 2.
3. Draw the arc away from the specified radius with the base point at the center of the circle.

Reply:
@amilcarlucas Have you looked at the images in OP? This has nothing to do with radius 0, it adds the possibility of center being the current position when you change to Circle mode instead of being *radius* meters in front of the vehicle.

Reply:
I looked at the images but I did not understood them.
The parameter description should be improved then, because with the current description that is not clear.